### ISSUE TITLE
Guice 4.0

### ISSUE THREAD

**Issue by fkranawetter** [englishtown/vertx-guice#6]
Would be cool to upgrade to Guice 4.0 as it has just been released officially including proper Java 8 support.

**Comment by adrianluisgonzalez** [comment]
Sure, are you using vert.x 2 or 3?

**Comment by fkranawetter** [comment]
Currently we require still vert.x 2, but we are going to port our project to vert.x 3 once the final release is available. Thanks a lot in advance! :)

**Comment by adrianluisgonzalez** [comment]
3.0.0-SNAPSHOT uses guice 4.0.  The feature branch can be seen here: https://github.com/englishtown/vertx-guice/tree/feature/guice-4

**Comment by jakub-bochenski** [comment]
Are there plans to release version 3?

**Comment by adrianluisgonzalez** [comment]
guice 4.0 was added in in commit [811d2421143f7da8a0aaa73057e0e28e89406c60](https://github.com/englishtown/vertx-guice/commit/811d2421143f7da8a0aaa73057e0e28e89406c60)

It was included with release 2.1.0

```xml
<dependency>
    <groupId>com.englishtown.vertx</groupId>
    <artifactId>vertx-guice</artifactId>
    <version>2.1.0</version>
</dependency>
```
